### PR TITLE
Libre2 Data Receiving from patched app

### PR DESCRIPTION
**Pull request by SandraK82**
Patched app instructions soon

### CODE REVIEW

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/DoNothingService.java [hunk 1]
```diff
@@ -114,7 +114,8 @@ public int onStartCommand(Intent intent, int flags, int startId) {
             }
         }
 
-        if (CollectionServiceStarter.isFollower(getApplicationContext())) {
+        if (CollectionServiceStarter.isFollower(getApplicationContext()) ||
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/DoNothingService.java]: What is the reason for attaching to the sync follower service? Wouldn't this implementation be closer to `NSEmulator` data collection type? I don't think any service is required.

File: app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CollectionServiceStarter.java [hunk 2]
```diff
@@ -116,6 +116,9 @@ private static void processPending() {
     public static boolean isFollower(Context context) {
         return Pref.getString("dex_collection_method", "").equals("Follower");
     }
+    public static boolean isLibre2App(Context context) {
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CollectionServiceStarter.java]: Are any of the changes in this class required?

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/DoNothingService.java (hunk 1 above)
Comment (SandraK82) [reply app/src/main/java/com/eveningoutpost/dexdrip/Services/DoNothingService.java]: Maybe I did not understand it correctly. I thougth , DexCollectionType::getCollectorServiceClass would return a DexCollectionsService anyways for NSEmulator. Therefore I used the DoNothingService instead

File: app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CollectionServiceStarter.java (hunk 2 above)
Comment (SandraK82) [reply app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CollectionServiceStarter.java]: yes for limitting the Calibrations (newer commit). If the DoNothingService is removed, private static boolean isLibre2App(String collection_method) can be removed too

Comment (SandraK82) [reply app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CollectionServiceStarter.java]: Calibrations are now in my repo too: https://github.com/SandraK82/xDrip-plus/commit/e3e13292f3319aaa7b8b37e1e29d7c593c073e0f

**Comment by SandraK82** [comment]
slope / intercept are now used ;)

**Comment by jamorham** [comment]
Can you give me an indication of how much testing has occurred with this code and previous iterations of it? Has this been used for full end to end reception of real data?

**Comment by SandraK82** [comment]
All four team members are testing without problems so far. I use it for looping since today

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java [hunk 3]
```diff
@@ -1241,22 +1241,50 @@ public static synchronized BgReading bgReadingInsertLibre2(double calculated_val
         // TODO slope!!
```
Comment (AdrianLxM) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java]: Does this slope refer to the calibration slope? If yes I believe the TODO could be removed :)

**Pull request merged**